Mathlib/Tactic/Tauto.lean

```diff
@@ -25,7 +25,21 @@ def distribNotOnceAt (hypFVar : Expr) (g : MVarId) : MetaM AssertAfterResult :=
   let .fvar fvarId := hypFVar | throwError "not fvar {hypFVar}"
   let h ← fvarId.getDecl
   let e : Q(Prop) ← (do guard <| ← Meta.isProp h.type; pure h.type)
-  let replace (p : Expr) := g.replace h.fvarId p
+  let replace (p : Expr) : MetaM AssertAfterResult := do
+    let s ← saveState
```

Comment:
I found a nicer way to handle this backtracking:
```lean
  let replace (p : Expr) : MetaM AssertAfterResult :=
    commitIfNoEx do
      let result ← g.assertAfter fvarId h.userName (← inferType p) p
      let newGoal ← result.mvarId.clear fvarId
      return { result with mvarId := newGoal }
```

Reply:
I pushed https://github.com/leanprover-community/mathlib4/pull/11799/commits/72c5f0a1188a4eb815dc3973c12c28e33d02e7cb with this change.